server/src/main/java/org/opensearch/index/compositeindex/datacube/ReadDimension.java

```diff
@@ -31,6 +32,22 @@ public String getField() {
         return field;
     }
 
+    @Override
+    public int getNumSubDimensions() {
+        return 1;
+    }
+
+    @Override
+    public int setDimensionValues(Long value, Long[] dims, int index) {
+        dims[index++] = value;
+        return index;
+    }
+
+    @Override
+    public List<String> getDimensionFieldsNames() {
+        return List.of(field);
```

Comment:
I'm assuming that each sub dimension translates to multiple ReadDimension objects?

Reply:
Yeah when we read, the sub dimensions are technically their own dimensions now. 

Timestamp { hour, minute } , gets saved and read as "timestamp_hour" and "timestamp_minute"